cloud-shell/cloudbuild.yaml

```diff
@@ -12,10 +12,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# The cloudbuild instructions originally generated by GCP to rebuild one image
+# look like below, for reference:
+
+# steps:
+# - name: 'gcr.io/cloud-builders/docker'
+#   args: ['build', '-t', 'gcr.io/$PROJECT_ID/$REPO_NAME', '.']
+# - name: 'gcr.io/cloudshell-images/custom-image-validation'
+#   args: ['image_test.py', '--image', 'gcr.io/$PROJECT_ID/$REPO_NAME']
+# images: ['gcr.io/$PROJECT_ID/$REPO_NAME']
+# timeout: '1200s'
+
 steps:
 - name: 'gcr.io/cloud-builders/docker'
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/$REPO_NAME', '.']
-- name: 'gcr.io/cloudshell-images/custom-image-validation'
-  args: ['image_test.py', '--image', 'gcr.io/$PROJECT_ID/$REPO_NAME']
-images: ['gcr.io/$PROJECT_ID/$REPO_NAME']
-timeout: '1200s'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |-
+    git clone https://github.com/GoogleCloudPlatform/cloud-ops-sandbox.git
+    cd cloud-ops-sandbox
+    for TAG in $(git tag); do
+        echo "checking out $$TAG"
+        git checkout $$TAG
+        # if cloud-build directory exists, build new container
+        if [[ -d "./cloud-shell" ]]; then
+            echo "building $$TAG"
+            docker build -t gcr.io/$PROJECT_ID/cloudshell-image:$$TAG ./cloud-shell
+            # the following step validates the image
+            docker run -t gcr.io/cloudshell-images/custom-image-validation image_test.py --image gcr.io/$PROJECT_ID/cloud-shell:$$TAG
```

Comment:
Does this work here? I assumed the image might need to be pushed before `gcr.io/cloudshell-images/custom-image-validation` could access it

Reply:
Looking in the Cloud Build logs, it's giving an error but not failing the pipeline:

```
OSError: [Errno 2] No such file or directory: '/workspace/tmpO1GCPI'
    _os.mkdir(file, 0700)
  File "/usr/lib/python2.7/tempfile.py", line 339, in mkdtemp
    tempdir = tempfile.mkdtemp(dir='/workspace')
  File "/bin/image_test.py", line 63, in temp_directory
    return self.gen.next()
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    with temp_directory() as tempdir:
  File "/bin/image_test.py", line 153, in container_session
    return self.gen.next()
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    with container_session('testuser') as session:
  File "/bin/image_test.py", line 172, in test_ssh
Traceback (most recent call last):
```

I'm not sure how important this validation step is. We might want to drop this, or add it in another PR later